### PR TITLE
[spark] Support ANALYZE table

### PR DESCRIPTION
**Pull request by Zouxxyy**
### Purpose

Support:

- ANALYZE TABLE T COMPUTE STATISTICS
- ANALYZE TABLE T COMPUTE STATISTICS FOR COLUMNS

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/spark/sql/Utils.scala [hunk 1]
```diff
@@ -68,4 +75,41 @@ object Utils {
...
+  }
+
+  def computeColumnStats(
+      sparkSession: SparkSession,
+      relation: LogicalPlan,
+      columns: Seq[Attribute]): (Long, Map[Attribute, ColumnStat]) = {
+    CommandUtils.computeColumnStats(sparkSession, relation, columns)
+  }
+
+  def analyzeSupportsType(dataType: DataType): Boolean = dataType match {
```
Comment (YannByron) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/spark/sql/Utils.scala:97]: Only these methods that are private in org.apache.spark.sql should be put here. Check these newly added ones.

File: paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java [hunk 2]
```diff
@@ -97,6 +98,18 @@ public AbstractFileStoreTable(
         this.catalogEnvironment = catalogEnvironment;
     }
 
+    @Override
+    public Optional<Stats> statistics() {
+        // todo: support time travel
+        if (coreOptions().startupMode().equals(CoreOptions.StartupMode.LATEST_FULL)) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java:104]: remove this, wrong stats also are useful for query.

File: paimon-core/src/main/java/org/apache/paimon/table/Table.java [hunk 3]
```diff
@@ -58,6 +59,9 @@ public interface Table extends Serializable {
     /** Optional comment of this table. */
     Optional<String> comment();
 
+    /** Optional statistics of this table. */
+    Optional<Stats> statistics();
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/Table.java:63]: add `@Experimental`

and add `Experimental` to Stats, ColStats. they are public api

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/PaimonAnalysis.scala [hunk 4]
```diff
@@ -49,6 +49,29 @@ case class PaimonPostHocResolutionRules(session: SparkSession) extends Rule[Logi
...
+          PaimonAnalyzeTableColumnCommand(
+            catalog,
+            identifier,
+            table,
+            Option.apply(Seq()),
+            allColumns = false)
+        }
+
+      case a @ AnalyzeColumn(
+            ResolvedTable(catalog: TableCatalog, identifier: Identifier, table: SparkTable, _),
```
Comment (YannByron) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/catalyst/analysis/PaimonAnalysis.scala:70]: ditto

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTest.scala [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+import org.apache.paimon.data.Decimal
+import org.apache.paimon.spark.PaimonSparkTestBase
+import org.apache.paimon.stats.ColStats
+import org.apache.paimon.utils.DateTimeUtils
+
+import org.apache.spark.sql.Row
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Assertions
+
+class AnalyzeTableTest extends PaimonSparkTestBase {
```
Comment (YannByron) [inline paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTest.scala:29]: Add tests for other spark versions to check the compatibility and applicability.

File: paimon-core/src/main/java/org/apache/paimon/table/Table.java (hunk 3 above)
Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/table/Table.java:63]: `Stats` can be `Statistics`

**Comment by YannByron** [comment]
+1 but left some comments. Looking forward to the next pr that should have the ability to improve spark query by paimon col statistics.

**Review by JingsongLi — APPROVED** [review]
+1

**Comment by Zouxxyy** [comment]
#2799

**Comment by YannByron** [comment]
https://github.com/apache/incubator-paimon/issues/2404

**Pull request merged**